File: packages/ts/lib/src/typescript/MerkleTree/Options.ts
import { SHA_256 } from '..'

/**
 * Define the `MerkleTree` options
 * 
 * @param {boolean} doubleHash - Set to `true` to hash nodes twice (default: `false`)
 * @param {string} engine - The name of the hashing function (default: `'sha-256'`)
 * @param {boolean} sort - Set to `true` to lexicographically sort leaves (default: `false`)
 */
export interface MerkleTreeOptions {
  doubleHash: boolean
  engine: string
  sort: boolean
}

export const MerkleTreeOptions = (doubleHash?: boolean, engine?: string, sort?: boolean): MerkleTreeOptions => ({
  doubleHash: doubleHash !== undefined ? doubleHash : false,
  engine: engine !== undefined ? engine : SHA_256,
  sort: sort !== undefined ? sort : false
})

File: packages/scala/src/main/scala/com/cyrildever/merkle/tree/Options.scala
package com.cyrildever.merkle.tree

import com.cyrildever.merkle.hash.Hash.SHA_256

/**
 * Define the `MerkleTree` options
 *
 * @author  Cyril Dever
 * @since   1.0
 * @version 1.0
 *
 * @param doubleHash Set to `true` to hash nodes twice (default: `false`)
 * @param engine The name of the hashing function (default: `"sha-256"`)
 * @param sort Set to `true` to lexicographically sort leaves (default: `false`)
 */
case class Options(doubleHash: Boolean = false, engine: String = SHA_256, sort: Boolean = false)
